kedro/io/core.py

```diff
@@ -567,6 +567,15 @@ def parse_dataset_definition(
         )
 
     dataset_type = config.pop(TYPE_KEY)
+
+    # This check prevents the use of dataset types with uppercase 'S' in 'Dataset',
+    # which is no longer supported as of Kedro 0.19.
+    if isinstance(dataset_type, str):
+        if dataset_type.endswith("Set"):
+            raise ValueError(
```

Comment:
Maybe this should be a warning since it could be a custom dataset too?

Reply:
hey after changing the code, I realised that we use the warning then everything continues running, but eventually, another error will arise because the DataSet is not found. In that case, why not display the error at this stage instead?

Reply:
yeah so if it the dataset doesn't exist in `kedro-datasets` or `kedro.io` then the proper error will show but if it's a custom dataset with the "DataSet" spelling, it should continue running and not error out.

Reply:
@ankatiyar hey got you, I've updated it to be a warning now and also reflects it in the test, to ensure it throws warning first, then eventually it will be error. Let me know what you think?